File: bootstack-service/src/main/java/com/bootstack/service/system/role/SystemRoleService.java
package com.bootstack.service.system.role; /**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 * <p>
 * http://www.apache.org/licenses/LICENSE-2.0
 * <p>
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import com.bootstack.model.page.PageModel;
import com.bootstack.model.system.role.SystemRoleModel;
import org.springframework.data.domain.Pageable;

/**
 * <p> SystemRoleService </p>
 * <p> Description : SystemRoleService </p>
 * <p> Author : qianmoQ </p>
 * <p> Version : 1.0 </p>
 * <p> Create Time : 2019-01-26 01:01 </p>
 * <p> Author Eamil: <a href="mailTo:shichengoooo@163.com">qianmoQ</a> </p>
 */
public interface SystemRoleService {

    /**
     * insert model
     *
     * @param model model info
     * @return insert result
     */
    Long insertModel(SystemRoleModel model);

    /**
     * get model by id
     *
     * @param id id
     * @return model response id
     */
    SystemRoleModel getModelById(Long id);

    /**
     * get all model
     *
     * @param pageable page info
     * @return all model
     */
    PageModel<SystemRoleModel> getAll(Pageable pageable);

    /**
     * get model by name
     *
     * @param name name
     * @return model response by name
     */
    SystemRoleModel getModelByName(String name);

}

File: core/authx-web/src/main/angular/src/services/system/system.role.service.ts
/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import {Injectable} from "@angular/core";
import {BaseService} from "../base.service";
import {Http, RequestOptions} from "@angular/http";
import {Router} from "@angular/router";
import {ToastyService} from "ng2-toasty";
import {Observable} from "rxjs";
import {CommonResponseModel} from "../../app/shared/model/common/response/response.model";
import {CommonPageModel} from "../../app/shared/model/common/response/page.model";
import {HttpUtils} from "../../app/shared/utils/http.util";
import {ApiConfig} from "../../config/api.config";
import {ResponseUtils} from "../../app/shared/utils/response.util";
import {SystemRoleParam} from "../../app/shared/param/system/role/system.role.param";
import {SystemMenuRoleParam} from "../../app/shared/param/system/menu/system.menu.role.param";
import {SystemRoleMenuParam} from "../../app/shared/param/system/role/system.role.menu.param";

/**
 * System Role Service
 */
@Injectable()
export class SystemRoleService implements BaseService {

    constructor(
        private http: Http,
        private options: RequestOptions,
        private router: Router,
        private toastyService: ToastyService) {
    }

    getInfo(primaryKey: Object): Observable<CommonResponseModel> {
        return undefined;
    }

    getList(page: CommonPageModel): Observable<CommonResponseModel> {
        const options = HttpUtils.getDefaultRequestOptionsByTokenAndJSON();
        const params = HttpUtils.getParams();
        params.append('page', page.number.toString());
        params.append('size', page.size.toString());
        options.params = params;
        return this.http.get(ApiConfig.API_SYSTEM_ROLE, options).map(ResponseUtils.extractData);
    }

    register(param: SystemRoleParam): Observable<CommonResponseModel> {
        const options = HttpUtils.getDefaultRequestOptionsByTokenAndJSON();
        return this.http.post(ApiConfig.API_SYSTEM_ROLE, JSON.stringify(param), options)
            .map(ResponseUtils.extractData);
    }

    update(param: SystemRoleParam): Observable<CommonResponseModel> {
        const options = HttpUtils.getDefaultRequestOptionsByTokenAndJSON();
        return this.http.put(ApiConfig.API_SYSTEM_ROLE, JSON.stringify(param), options)
            .map(ResponseUtils.extractData);
    }

    /**
     * get menu tree list by role and type
     * @param page
     */
    getTreeListByRoleAndType(param: SystemMenuRoleParam): Observable<CommonResponseModel> {
        const options = HttpUtils.getDefaultRequestOptionsByTokenAndJSON();
        const params = HttpUtils.getParams();
        params.append('role', param.role.toString());
        params.append('menuType', param.menuType.toString());
        options.params = params;
        return this.http.get(ApiConfig.API_SYSTEM_ROLE_TREE_LIST, options).map(ResponseUtils.extractData);
    }

    updateRoleMenu(param: SystemRoleMenuParam): Observable<CommonResponseModel> {
        const options = HttpUtils.getDefaultRequestOptionsByTokenAndJSON();
        return this.http.put(ApiConfig.API_SYSTEM_ROLE_TREE, JSON.stringify(param), options).map(ResponseUtils.extractData);
    }

}
